### DISCUSSION TITLE
Regarding governance

### DISCUSSION THREAD

**Discussion by wjk** [microsoft/WindowsAppSDK#1907] [Feedback wanted]
> this is something we'll need to follow up with an internal deliverable

I was just told this in https://github.com/microsoft/terminal/issues/11916. I am not amused that this has become such an incredibly common, even stock response in Microsoft’s open-source projects these days. I know you can’t release private Windows APIs, but this is becoming ridiculous. (And if it’s because third-party use of those APIs would pose a security risk, then you’re relying on security through obscurity, which is never a good idea.)

What happened to detaching new features from the OS to ease downlevel compatibility? You promised us that you would do this many times, but then you go and outright ignore it. (If a bug needs a fix in the OS to work properly, then software encountering that bug requires use of the very-latest Windows build. This requirement is even more onerous if the build in question is Insider-only. Remember, Windows is not exclusively intended for use by experienced software developers who are enrolled in the Insider program.) What happened to guiding projects based on community input? Everyone’s feedback is disregarded because because some “large private partner” is always running into some minor blocker that must be taken care of immediately, never mind those huge design flaws that everyone else is constantly complaining about. Hell, what happened to releasing the source code at all? I am quite sick of this sort of conduct from MS.

(edit — remove vitriol)

**Comment by riverar** [comment]
How does this Terminal issue relate to this repository?

**Reply by wjk** [reply]
It was the straw that broke the camel’s back. I should have made that more clear. Sorry!

**Reply by wjk** [reply]
And I thought it would get more visibility here.

**Reply by wjk** [reply]
If you don’t mind me asking, @riverar, are you a Microsoft employee? Your profile does not make this clear.

**Reply by riverar** [reply]
I am not. Just your neighborhood stray dog/Windows Development MVP 😄. You're free to ignore anything I say!
